xfields/beam_elements/electroncloud.py

```diff
@@ -0,0 +1,148 @@
+            remove closed orbit distortion effects.
+        dipolar_ptau_kick (float): subtracts a constant value from the kick to 
+            ptau. Should be set equal to the field map's kick at the closed 
+            orbit to remove closed orbit distortion effects.
+        length (float): the length of the electron-cloud interaction in
+            meters.
+        apply_z_kick (bool): If ``True``, the longitudinal kick on the
+            particles is applied. The default is ``True``.
+        fieldmap (xfields.TriCubicInterpolatedFieldMap): Field map of the 
+            electron cloud forces.
+    Returns:
+        (ElectronCloud): An electron cloud beam element.
+    """
+
+    _xofields = {
+        'x_shift': xo.Float64,
+        'y_shift': xo.Float64,
+        'tau_shift': xo.Float64,
+        'dipolar_px_kick': xo.Float64,
+        'dipolar_py_kick': xo.Float64,
+        'dipolar_ptau_kick': xo.Float64,
+        'length': xo.Float64,
+        #'fieldmap': TriCubicInterpolatedFieldMapData,
+        'fieldmap': xo.Ref(TriCubicInterpolatedFieldMapData),
+        }
+
+#     def copy(self, _context=None, _buffer=None, _offset=None):
+#         if _buffer is not self._buffer:
+#             raise NotImplementedError
+#         return SpaceCharge3D(_context=_context,
+#                 _buffer=_buffer, _offset=_offset,
+#                 update_on_track=self.update_on_track,
+#                 length=self.length,
+#                 apply_z_kick=self.apply_z_kick,
+#                 fieldmap=self.fieldmap)
+
+    def __init__(self,
+                 _context=None,
+                 _buffer=None,
+                 _offset=None,
+                 x_shift=0.,
+                 y_shift=0.,
+                 tau_shift=0.,
+                 dipolar_px_kick=0.,
+                 dipolar_py_kick=0.,
+                 dipolar_ptau_kick=0.,
+                 length=None,
+                 apply_z_kick=True,
+                 fieldmap=None,
+                 ):
+
+        # To be implemented if false
+        self.apply_z_kick = apply_z_kick
+
+        if _buffer is not None:
+            _context = _buffer.context
+        if _context is None:
+            _context = xo.context_default
+
+        # if fieldmap is None:
+        #     # I build the fieldmap on a temporary buffer
+        #     temp_buff = _context.new_buffer()
+        #     fieldmap = TriLinearInterpolatedFieldMap(
+        #                 _buffer=temp_buff,
+        #                 rho=rho, phi=phi,
+        #                 x_grid=z_grid, y_grid=y_grid, z_grid=z_grid,
+        #                 x_range=x_range, y_range=y_range, z_range=z_range,
+        #                 dx=dx, dy=dy, dz=dz,
+        #                 nx=nx, ny=ny, nz=nz,
+        #                 solver=solver,
+        #                 scale_coordinates_in_solver=scale_coordinates_in_solver,
+        #                 updatable=update_on_track,
+        #                 fftplan=fftplan)
```

Comment:
What happens if  ```fieldmap``` is None?

Reply:
c7a5b50

For now assert that fieldmap is not None. Might limit some flexibility as it forces the user to create the fieldmap before creating the element. To be revisited if ever needed to create fieldmap after creating the element.